views/js/tools/diagnostic/diagnostic.js

```diff
@@ -484,15 +485,20 @@ define([
                 }
             });
         }
+        this.config = getConfig(config, _defaults);
```

Comment:
This line is incorrect. The `this` keyword cannot be used here, we are in a factory, not in an object.
Then this is not the place to init the config. The default values are set to the `component` factory as this is the second parameter: `component(specs, defaults)`.

Then the config must be passed to the `init()` method.